File: Server/team05_server_source/Server/src/main/java/de/team05/server/json/MatchConfigJson.java
package de.team05.server.json;

public class MatchConfigJson {

    public Integer maxRounds;
    public TimeoutsJson timings;
    public ProbabilitiesJson probabilities;

    public class TimeoutsJson {
        public Integer teamFormationTimeout;
        public Integer playerTurnTimeout;
        public Integer fanTurnTimeout;
        public Integer minPlayerPhaseAnimationDuration;
        public Integer minFanPhaseAnimationDuration;
        public Integer minBallPhaseAnimationDuration;
    }

    public class ProbabilitiesJson {
        public Float throwSuccess;
        public Float knockOut;
        public Float foolAway;
        public Float catchSnitch;
        public Float catchQuaffle;
        public Float wrestQuaffle;
        public ExtraMoveJson extraMove;
        public FoulDetectionJson foulDetection;
        public FanFoulDetectionJson fanFoulDetection;
    }

    public class ExtraMoveJson {
        public Float tinderblast;
        public Float cleansweep11;
        public Float comet260;
        public Float nimbus2001;
        public Float firebolt;
    }

    public class FoulDetectionJson {
        public Float flacking;
        public Float haversacking;
        public Float stooging;
        public Float blatching;
        public Float snitchnip;
    }

    public class FanFoulDetectionJson {

        public Float elfTeleportation;
        public Float goblinShock;
        public Float trollRoar;
        public Float snitchSnatch;
        public Float wombatPoo;
    }
}

File: Client/Client/Assets/Scripts/JsonScripts/MatchConfigJson.cs
using System;

[Serializable]
public class MatchConfigJson
{
    public int maxRounds;
    public TimeoutsJson timings;
    public ProbabilitiesJson probabilities;

    [Serializable]
    public class TimeoutsJson
    {
        public int teamFormationTimeout;
        public int playerTurnTimeout;
        public int fanTurnTimeout;
        public int minPlayerPhaseAnimationDuration;
        public int minFanPhaseAnimationDuration;
        public int minBallPhaseAnimationDuration;
    }

    [Serializable]
    public class ProbabilitiesJson
    {
        public float throwSuccess;
        public float knockOut;
        public float foolAway;
        public float catchSnitch;
        public float catchQuaffle;
        public float wrestQuaffle;
        public ExtraMoveJson extraMove;
        public FoulDetectionJson foulDetection;
        public FanFoulDetectionJson fanFoulDetection;
    }

    [Serializable]
    public class ExtraMoveJson
    {
        public float tinderblast;
        public float cleansweep11;
        public float comet260;
        public float nimbus2001;
        public float firebolt;
    }

    [Serializable]
    public class FoulDetectionJson
    {
        public float flacking;
        public float haversacking;
        public float stooging;
        public float blatching;
        public float snitchnip;
    }

    [Serializable]
    public class FanFoulDetectionJson
    {
        public float elfTeleportation;
        public float goblinShock;
        public float trollRoar;
        public float snitchSnatch;
    }
}
